contracts/src/systems/combat/contracts.cairo

```diff
@@ -0,0 +1,398 @@
+#[dojo::contract]
+mod combat_systems {
+    use eternum::alias::ID;
+
+    use eternum::models::resources::ResourceCost;
+    use eternum::models::position::{Position, HomePosition};
+    use eternum::models::config::SpeedConfig;
+    use eternum::models::movable::{Movable, ArrivalTime};
+    use eternum::models::owner::Owner;
+    use eternum::models::quantity::Quantity;    
+    use eternum::models::combat::{
+        Health, Attack, Defence, Duty, TownWatch
+    };
+
+    use eternum::systems::combat::interface::{
+        ISoldierSystems
+    };
+
+    use eternum::constants::{
+        WORLD_CONFIG_ID, SOLDIER_ENTITY_TYPE
+    };
+
+
+    #[external(v0)]
+    impl SoldierSystemsImpl of ISoldierSystems<ContractState> {
+
+        fn create_soldiers( 
+            self: @ContractState, world: IWorldDispatcher, 
+            entity_id: u128, quantity: u128
+        ) {
+
+            // q: check if entity is a realm?
+
+            let caller = starknet::get_caller_address();
+
+            let entity_owner = get!(world, entity_id, Owner);
+            assert(
+                entity_owner.address == caller,
+                    'not entity owner'
+            );
+
+            let entity_position = get!(world, entity_id, Position);
+
+            // todo@credence add payment here
+
+            let mut index = 0;
+            loop {
+                if index == quantity {
+                    break;
+                }
+
+                let soldier_id: u128 = world.uuid().into();
+                set!(world, (
+                    Owner {
+                        entity_id: soldier_id,
+                        address: caller
+                    },
+                    Health {
+                        entity_id: soldier_id,
+                        value: 100
+                    },
+                    Attack {
+                        entity_id: soldier_id,
+                        value: 100
+                    },
+                    Defence {
+                        entity_id: soldier_id,
+                        value: 100
+                    },
+                    Quantity {
```

Comment:
does this need to exist?

Reply:
Hmm I think so. Quantity value is the only way we can differentiate between single soldiers and groups of them (like Raiders)

Reply:
ok right - i understand now